src/Monetization.php

```diff
@@ -406,4 +420,49 @@ public function getPrepaidBalanceReports(string $developer_id, \DateTimeImmutabl
       ->getReport($month, $currency);
   }
 
+  /**
+   * {@inheritdoc}
+   */
+  public function formUserAdminPermissionsAlter(&$form, FormStateInterface $form_state, $form_id) {
+    // Disable All incompatible permissions in the UI.
+    foreach (array_keys($this->getMonetizationPermissions()) as $permission_name) {
+      if (isset($form['permissions'][$permission_name]['anonymous'])) {
+        // Disable the permission.
+        $form['permissions'][$permission_name]['anonymous']['#disabled'] = TRUE;
+        $form['permissions'][$permission_name]['anonymous']['#value'] = 0;
+      }
+    }
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function userRolePresave(RoleInterface $user_role) {
+    // This prevents monetization permission grants on config import.
+    if ($user_role->id() === 'anonymous') {
+      // Get any permissions anon shouldn't have.
+      $unauthorized_perms = array_intersect($user_role->getPermissions(), array_keys($this->getMonetizationPermissions()));
+      // Remove all unauthorized perms.
+      foreach ($unauthorized_perms as $permission_name) {
+        $user_role->revokePermission($permission_name);
+        $this->logger->info('Removing the `%permission` permission from the anonymous role.', ['%permission' => $permission_name]);
+      }
+    }
+
+  }
+
+  /**
+   * Gets a list of Apigee monetization permissions.
+   *
+   * These permissions should only be applied to authenticated roles.
+   *
+   * @return array
+   *   Permissions for the monetization module.
+   */
+  protected function getMonetizationPermissions() {
+    return array_filter($this->permission_handler->getPermissions(), function ($permission) {
+      return ($permission['provider'] === 'apigee_m10n');
```

Comment:
Do we need to handle m10n submodules? I know we're going to have a few permissions in `apigee_m10n_add_credit`.

Reply:
Yes, but those will be follow-ups to this.